bfkrickard/MusicGame
# Redux implementaiton

## Redux/flux pattern
* UI calls an action > action dispatches to all reducers > reducer updates the data in the `store`
    * ![image](https://user-images.githubusercontent.com/3317835/94330017-0b0ed780-ff75-11ea-9bef-40301e7fec43.png)


## Updates
**Update**:
- added redux for state management
   - Installed the required dependencies (see `package.json`)
   - allows us to have 1 source of truth for data storage
   - access the data from anywhere within the app

* `src/store/index.js`
  * we initialize the global store here
  * this gets passed in to the app on initial start (see `App.js`)
- `src/actions/login.js`
   - setup first action for login
   - `Dispatch` the action to the reducers

- `/src/reducers/*`
    - setup first reducer for user

- `/src/constants/*`
   - added constants for action types

**fixes**:
- fixed jsconfig.json so intellisense for aliases will work properly

## Testing
* Download this to your chrome browser https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd

* pull branch
* run `yarn install` - to install latest dependencies
* run `yarn web` - start the app
* if you've installed the chrome dev tool you should see it turn on automatically 
    * ![image](https://user-images.githubusercontent.com/3317835/94330390-ffbdab00-ff78-11ea-8dab-08a4fe4ef183.png)

* enter a name in the textbox and hit the spotify button
* see that the store contains the name you entered
    * ![image](https://user-images.githubusercontent.com/3317835/94331189-3e566400-ff7f-11ea-94ae-68ae0c45a087.png)
     * This is the 'global' storage
